### PR TITLE
Add uniqueId to Person model

### LINKED ISSUE #63: Add unique id to Person model

### PR DESCRIPTION
**Pull request by conradsoon**
This fixes #63

### CODE REVIEW

File: src/main/resources/view/PersonListCard.fxml [hunk 1]
```diff
@@ -27,6 +27,10 @@
         </Label>
         <Label fx:id="name" text="\$first" styleClass="cell_big_label" />
       </HBox>
+      <HBox spacing="5" alignment="CENTER_LEFT">
+        <Label text="uniqueID:" styleClass="cell_small_label" />
```
Comment (lordidiot) [inline src/main/resources/view/PersonListCard.fxml]: uniqueID is quite verbose, I think we should stick to `id`

**Review by lordidiot — CHANGES_REQUESTED** [review]
Thanks for working on this @conradsoon. The current implementation looks functional but I think it has weak abstraction.

Relying on `AddCommand` or other users of `Person` to set the id can lead to bugs in future when other functionalities create `Person`'s but do not set the `id` accordingly.

Instead, I'd prefer if `AddressBook` takes responsibility to set the ids in `AddressBook::addPerson`. The `getNextId` can then be kept as an private implementation detail in AddressBook that does not need to be exposed through `Model`. In this case, even `Person::setId` can be made to be package-private instead of public to abstract further.

Happy to discuss more about this.

File: src/main/resources/view/PersonListCard.fxml (hunk 1 above)
Comment (conradsoon) [reply src/main/resources/view/PersonListCard.fxml]: Hey Calvin, good points raised. Agree mostly with your points, will change it accordingly. Thanks.

**Comment by lordidiot** [comment]
One more note @conradsoon, I don't think this current PR allows for the ids to persist. Please do so by editing `JsonAdaptedPerson` accordingly.

**Comment by conradsoon** [comment]
@lordidiot Have made the changes requested, think logic of ID handling is much clearer now. Thanks for the code review.

File: src/main/java/seedu/address/model/Model.java [hunk 2]
```diff
@@ -98,4 +98,8 @@ public interface Model {
      * @return List of events.
      */
     List<Event> getEventList();
+
```
Comment (lordidiot) [inline src/main/java/seedu/address/model/Model.java]: Leftover comments

File: src/main/java/seedu/address/model/person/Person.java [hunk 3]
```diff
@@ -129,6 +132,14 @@ public Set<Tag> getEmergencyTags() {
             .collect(Collectors.toSet());
     }
 
+    public Optional<Integer> getId() {
+        return id;
+    }
```
Comment (lordidiot) [inline src/main/java/seedu/address/model/person/Person.java:136]: nit: space between functions

File: src/main/java/seedu/address/model/util/SampleDataUtil.java
Comment (lordidiot) [inline src/main/java/seedu/address/model/util/SampleDataUtil.java:1]: Please give the sample Person instances each a unique id, a valid state contains Persons all with ids.

File: src/test/java/seedu/address/model/AddressBookTest.java [hunk 4]
```diff
@@ -83,12 +85,42 @@ public void getPersonList_modifyList_throwsUnsupportedOperationException() {
         assertThrows(UnsupportedOperationException.class, () -> addressBook.getPersonList().remove(0));
     }
 
+    @Test
+    public void randomOperations_ensureUniqueIDs() {
```
Comment (lordidiot) [inline src/test/java/seedu/address/model/AddressBookTest.java]: `IDs` -> `Ids` to match our [coding style](https://se-education.org/guides/conventions/java/intermediate.html#:~:text=Abbreviations%20and%20acronyms%20should%20not%20be%20uppercase%20when%20used%20as%20a%20part%20of%20a%20name.)

**Review by lordidiot — APPROVED** [review]
LGTM! Thanks for the changes. Left a few nitpicks and one important comment on `SampleDataUtil.java`. Once those changes are made please go ahead and merge this!

**Pull request merged**